examples/extras_provider.py

```diff
@@ -39,9 +39,9 @@ def get_base_requirement(self, candidate):
         """
         raise NotImplementedError
 
-    def identify(self, dependency):
-        base = super(ExtrasProvider, self).identify(dependency)
-        extras = self.get_extras_for(dependency)
+    def identify(self, requirement_or_candidate):
```

Comment:
Eww, is that *really* the best name we can use here? I wasn't that keen on "dependency" but even so...

I've always had a problem talking about this because I don't know what to call things. In pip terms, it's basically the "name" of the "project" that the requirement or candidate refers to, but neither of those terms are used by resolvelib, and I'm not sure what would be best as more neutral terms (or even if we need terms to be more neutral - at some point I feel that we're making it harder to reason about resolvelib because we're avoiding naming some of the concepts in order to be more generic...

But that's a lot of worrds for what is ultimately a fairly minor point, so feel free to ignore me...

Reply:
> Eww, is that _really_ the best name we can use here?

I mean, I'm sure we're open to other names. This is a actually-not-completely-horrible stop-gap solution until one of us has a wonderfully good brain wave. :P

Reply:
I really should look into adding type hints… We could as well call this `item` (or whatever generic name) if it is type-hinted as `Union[Candidate, Requirement]`.

Reply:
Heads up -- .pyi files (and py.typed) would likely be ideal. :)